### PR TITLE
plugins.cbsnews: support for live streams from CBS News

### PR DESCRIPTION
**Pull request by beardypig**
Support for live streams from CBS News.

Including:
- https://www.cbsnews.com/live/
- https://www.cbsnews.com/live/cbsn-local-bay-area/
- https://www.cbsnews.com/live/cbsn-local-ny/

### CODE REVIEW

File: src/streamlink/plugins/cbsnews.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from __future__ import print_function
```
Comment (bastimeyer) [inline src/streamlink/plugins/cbsnews.py:1]: There is no `print` call in the module, so the `__future__` import is unnecessary.

File: src/streamlink/plugins/cbsnews.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def _get_streams(self):
+        res = self.session.http.get(self.url)
+        _re_json = re.compile(r"CBSNEWS.defaultPayload = (\{.*)")
+        m = _re_json.search(res.text)
+        if m:
+            data = parse_json(m.group(1))
+            for i in data['items']:
+                for s in HLSStream.parse_variant_playlist(self.session, i['video']).items():
+                    yield s
```
Comment (bastimeyer) [inline src/streamlink/plugins/cbsnews.py:25]: The HLS URL should ideally be extracted via a validation schema.

```suggestion
_re_default_payload = re.compile(r"CBSNEWS.defaultPayload = (\{.*)")

_schema_items = validate.Schema(
    _re_default_payload.search,
    validate.any(None, validate.all(
        validate.get(1),
        validate.transform(parse_json),
        {"items": [validate.all({
            "video": validate.url(),
            "format": "application/x-mpegURL"
        }, validate.get("video"))]},
        validate.get("items")
    ))
)

def _get_streams(self):
    items = self.session.get(self.url, schema=self._schema_items)
    if items:
        for hls_url in items:
            for s in HLSStream.parse_variant_playlist(self.session, hls_url).items():
                yield s
```

Also please don't mix quotation styles.

File: tests/plugins/test_cbsnews.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import unittest  # noqa: F401
+import pytest
+from streamlink.plugins.cbsnews import CBSNews
+
+
+class TestCBSNews:
```
Comment (bastimeyer) [inline tests/plugins/test_cbsnews.py:6]: What is the reason for not using `unittest.TestCase`?
We shouldn't be mixing test setups. Almost all tests are written with unittest and pytest is only used as the test runner. Mixing this makes it much more difficult to read and maintain.

**Review by bastimeyer — CHANGES_REQUESTED** [review]

File: tests/plugins/test_cbsnews.py (hunk 3 above)
Comment (beardypig) [reply tests/plugins/test_cbsnews.py:6]: This has been used previously; we have `pytest` available and it has superior test builder functions.

File: src/streamlink/plugins/cbsnews.py (hunk 2 above)
Comment (beardypig) [reply src/streamlink/plugins/cbsnews.py:25]: I did not feel it useful or necessary to use the validation methods.  
Bit nit-picky on the quotes, but OK.

**Review by bastimeyer — CHANGES_REQUESTED** [review]
Sorry about the previous diff. I had only written it as a rather simple comment without actually testing it and it was not meant to be copied as a whole, just a suggestion.

File: src/streamlink/plugins/cbsnews.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
+
+
+class CBSNews(Plugin):
+    url_re = re.compile(r"https://www\.cbsnews\.com/live/")
+    _re_default_payload = re.compile(r"CBSNEWS.defaultPayload = (\{.*)")
+
+    _schema_items = validate.Schema(
+        _re_default_payload.search,
```
[the remark this answers is not in the document]
Comment (beardypig) [reply src/streamlink/plugins/cbsnews.py:14]: Ah yeah, already fix that but didn't push it :sigh:

**Review by bastimeyer — APPROVED** [review]

**Pull request merged**